src/server/common.cc

```diff
@@ -240,6 +240,18 @@ TieredStats& TieredStats::operator+=(const TieredStats& o) {
   return *this;
 }
 
+SearchStats& SearchStats::operator+=(const SearchStats& o) {
+  static_assert(sizeof(SearchStats) == 24);
+  ADD(used_memory);
+  ADD(num_entries);
+
+  DCHECK(num_indices == 0 || num_indices == o.num_indices);
+  num_indices = std::max(num_indices, o.num_indices);
```

Comment:
I'm confused, where is this set to a value != 0? 
Ctrl-f shows the test expects this to be 2, but I can't find where this value is set to other values?

Reply:
That's only the += operator. Its set to the indices count in the created stats structs. Because I need to sum it first with a defualt-intialized value, I check for == 0 as well

Reply:
I could store a optional<> by default to not check for == 0 and always check that those two are equal, but...

Reply:
I realize this is the += operator, but with ctrl-f I couldn't find where it is set to the index count

Reply:
I don't think there's anything wrong with this approach, I just think I'm missing something

Reply:
In `ShardDocIndices::GetStats()`

Reply:
Gotcha, that's why ctrl-f didn't help :)
Side note, I think it's clearer and less bug prone to use the C++20/C-whatever syntax of {.field = value, .field2 = value2}, but perhaps it's just me

Reply:
We do it in some places, but its C++20 technically (but all compilers support it with older versions of the standard)... 🤯  I like it more but avoid it for small number of fields 🤷🏻‍♂️

Reply:
Yeah, it's C++20, but because it's in C99 (it took 21 years to bring it to C++!) all compilers support it
I'd actually use it everywhere, no matter how many fields, but.. your call